astropy/units/tests/test_registry.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+
+import astropy.io.fits as fits
+import astropy.units as u
+from astropy.utils.data import get_pkg_data_filename
+
+
+@pytest.mark.remote_data
```

Comment:
I do like the very explicit test, but think this one may make more sense in `io.fits` (inside `units`, I'd prefer not to use anything else from astropy).  Aside: if it is from the astropy server, you can use `remote_data(source="astropy")`. Or maybe better to get just the header as @pllim suggests (the name suggests this is a large file!).

I'd then add some very simple tests just using `u.Unit('counts')`, like @pllim suggested.

For the location, you could just use `test_units.py`. That file is where `add_enabled_units` is tested as well - though "tested" is a bit of an overstatement with just one or two tests! And `set_enabled_units` is not tested explicitly at all...

Alternatively, since `test_units.py` is very long and there are rather few registry tests in `test_units.py` anyway, you could move those tests here.